### PR TITLE
GSoC 2019: Integrates Primitives with Existing UnitTest module

### PR DESCRIPTION
**Pull request by krishnadey30**
This PR integrates the primitives `gather tests` and `get test by index` with the existing UnitTest Module.
This also removes the method `runTest` which used to take `FCF`'s as argument as adds a method `main`. 
Now user can simply call `UnitTest.main()` for running all the test functions.

### CODE REVIEW

File: test/library/packages/UnitTest/UnitTest.chpl [hunk 1]
```diff
@@ -1081,18 +1081,14 @@ module UnitTest {
...
+    pragma "unsafe"
+    proc addTest(test) {
       this._tests.push_back(test);
       this.testCount += 1;
     }
 
-    proc addTests(tests) lifetime this < tests {
-      /*if isString(tests) then
-        compilerError("tests must be an iterable, not a string");*/
+    pragma "unsafe"
```
Comment (ben-albrecht) [inline test/library/packages/UnitTest/UnitTest.chpl]: I think we should add a comment explaining why we have these.

Comment (krishnadey30) [reply test/library/packages/UnitTest/UnitTest.chpl]: Okay Sure.

**Comment by krishnadey30** [comment]
> ```chapel
> SomeModuleContainingTests
> ```

Adding a test for the same.

File: test/library/packages/UnitTest/UnitTest.chpl (hunk 1 above)
Comment (krishnadey30) [reply test/library/packages/UnitTest/UnitTest.chpl]: I have updated it. Thanks

**Comment by ben-albrecht** [comment]
@krishnadey30 - Going to test & merge this now that https://github.com/chapel-lang/chapel/pull/13768 is merged.

**Comment by ben-albrecht** [comment]
Testing output:

```
Future (bug: == in LinkedList is giving false even if both are having same data) [Error matching program output for library/packages/UnitTest/AssertEqual/EqualLinkedList]
Future (bug: `canResolve` not suppporting unbounded ranges) [Error matching compiler output for library/packages/UnitTest/AssertGreater/UnEqualRangeGreater]
Future (Feature Request: Class methods should be considered as tests) [Error matching program output for library/packages/UnitTest/PopulateTests_PRIM/TestCountClass]
[Summary: #Successes = 16 | #Failures = 0 | #Futures = 3 | #Warnings = 0 ]
```

👍

**Comment by bradcray** [comment]
@krishnadey30, @ben-albrecht: I seem to be seeing a failure in testing on master.  Could it be due to this PR?

```
[Executing diff launcher_Test.numLocales.comm-none.good Launcher_Test.0-6.exec.out.tmp]
36,45d35
< SKIPPED test_locales.chpl: s14()
< ----------------------------------------------------------------------
< s14() skipped as s10() gave an Error
<
< ======================================================================
< SKIPPED test_locales.chpl: s11()
< ----------------------------------------------------------------------
< s11() skipped as s14() skipped
<
< ======================================================================
63a54,63
> SKIPPED test_locales.chpl: s14()
> ----------------------------------------------------------------------
> s14() skipped as s10() gave an Error
>
> ======================================================================
> SKIPPED test_locales.chpl: s11()
> ----------------------------------------------------------------------
> s11() skipped as s14() skipped
>
> ======================================================================
[Error matching program output for library/packages/UnitTest/Launcher_Test (execopts: 6)]
```

**Comment by krishnadey30** [comment]
> @krishnadey30, @ben-albrecht: I seem to be seeing a failure in testing on master. Could it be due to this PR?
> 
> ```
> [Executing diff launcher_Test.numLocales.comm-none.good Launcher_Test.0-6.exec.out.tmp]
> 36,45d35
> < SKIPPED test_locales.chpl: s14()
> < ----------------------------------------------------------------------
> < s14() skipped as s10() gave an Error
> <
> < ======================================================================
> < SKIPPED test_locales.chpl: s11()
> < ----------------------------------------------------------------------
> < s11() skipped as s14() skipped
> <
> < ======================================================================
> 63a54,63
> > SKIPPED test_locales.chpl: s14()
> > ----------------------------------------------------------------------
> > s14() skipped as s10() gave an Error
> >
> > ======================================================================
> > SKIPPED test_locales.chpl: s11()
> > ----------------------------------------------------------------------
> > s11() skipped as s14() skipped
> >
> > ======================================================================
> [Error matching program output for library/packages/UnitTest/Launcher_Test (execopts: 6)]
> ```

Hey @bradcray I will fix this. This is due to not having a good file for `CHPL_COMM=none`

**Comment by krishnadey30** [comment]
> @krishnadey30, @ben-albrecht: I seem to be seeing a failure in testing on master. Could it be due to this PR?
> 
> ```
> [Executing diff launcher_Test.numLocales.comm-none.good Launcher_Test.0-6.exec.out.tmp]
> 36,45d35
> < SKIPPED test_locales.chpl: s14()
> < ----------------------------------------------------------------------
> < s14() skipped as s10() gave an Error
> <
> < ======================================================================
> < SKIPPED test_locales.chpl: s11()
> < ----------------------------------------------------------------------
> < s11() skipped as s14() skipped
> <
> < ======================================================================
> 63a54,63
> > SKIPPED test_locales.chpl: s14()
> > ----------------------------------------------------------------------
> > s14() skipped as s10() gave an Error
> >
> > ======================================================================
> > SKIPPED test_locales.chpl: s11()
> > ----------------------------------------------------------------------
> > s11() skipped as s14() skipped
> >
> > ======================================================================
> [Error matching program output for library/packages/UnitTest/Launcher_Test (execopts: 6)]
> ```

Have opened a PR https://github.com/chapel-lang/chapel/pull/13807

**Pull request merged**